kingpin/actors/aws/ecs.py

```diff
@@ -0,0 +1,242 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2014 Nextdoor.com, Inc
+
+"""
+:mod:`kingpin.actors.aws.ecs`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+"""
+
+import json
+import logging
+
+from tornado import gen
+
+from kingpin import utils
+from kingpin.actors import exceptions
+from kingpin.actors.aws import base
+from kingpin.actors.utils import dry
+from kingpin.constants import REQUIRED
+
+log = logging.getLogger(__name__)
+
+__author__ = 'Steve Mostovoy <smostovoy@nextdoor.com>'
+
+
+class ECSBaseActor(base.AWSBaseActor):
+
+    """Base class for ECS actors."""
+
+    all_options = {
+        'region': (str, REQUIRED,
+                   'AWS region (or zone) name, like us-west-2 or eu-west-1'),
+        'cluster': (str, REQUIRED, 'ECS cluster in the given region to target')
+    }
+
+    def _parse_task_definition_file(self, task_definition_file, parameters):
+        """Parse and interpolate variables into the task definition file.
+
+        Args:
+            task_definition_file: Local path to the file to parse.
+            parameters: Dictionary of parameters to interpolate.
+
+        Returns:
+            Python object with the parsed results.
+        """
+        task_definition_file = task_definition_file
+        body = self.readfile(task_definition_file)
+        body = utils.populate_with_tokens(body, parameters)
+        return json.loads(body)
+
+    def _handle_failures(self, failures):
+        """Logs ECS failures. Raises an unrecoverable exception if there are any.
+
+        Args:
+            failures: list of failures from ECS api.
+        """
+        # TODO: When can we recover? What are the failures? How do we recover?
+        if not failures:
+            return
+        for failure in failures:
+            self.log.error(failure)
+        raise exceptions.UnrecoverableActorFailure()
+
+
+class RunTask(ECSBaseActor):
+
+    """Register and run a task on ECS.
+
+    This actor will loop indefinitely until the task is complete.
+
+    **Options**
+
+    :task_definition:
+      String of path to the Task Definition file template
+      Must be a local file path
+      Parameters to be interpolated must be of the form %VAR%
+
+    :parameters:
+      A dictionary of key/value pairs used to fill in the parameters for the
+      Task Definition template
+
+    :region:
+      AWS region (or zone) name, such as us-west-2 or eu-west-1
+
+    :cluster:
+      ECS cluster in the region to launch the task into
+
+    :count:
+      How many tasks to run
+
+    :skip_wait:
```

Comment:
I'd re-name this to `wait_for_completion`, or even just `wait` and default it to `True`.